service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpResponseVerification.java

```diff
@@ -6,53 +6,41 @@
 package com.microsoft.azure.sdk.iot.service.transport.amqps;
 
 import com.microsoft.azure.sdk.iot.service.exceptions.*;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.qpid.proton.amqp.Symbol;
 import org.apache.qpid.proton.amqp.messaging.*;
 import org.apache.qpid.proton.amqp.transport.AmqpError;
 import org.apache.qpid.proton.amqp.transport.DeliveryState;
 import org.apache.qpid.proton.amqp.transport.ErrorCondition;
 
+@Slf4j
 public class AmqpResponseVerification implements AmqpError
 {
     private Symbol errorCondition = null;
     private String errorDescription = null;
-    //Codes_SRS_SERVICE_SDK_JAVA_AMQPRESPONSEVERIFICATION_25_006: [** The function shall save null exception if the amqp delivery state is accepted or received or released or modified **]**
     private IotHubException exception = null;
 
     public AmqpResponseVerification(DeliveryState state)
     {
         if (state == null)
         {
-            //Codes_SRS_SERVICE_SDK_JAVA_AMQPRESPONSEVERIFICATION_25_008: [** The function shall save IotHubException if the amqp delivery state is null or undefined as per AMQP spec. **]**
             this.exception = new IotHubException();
+            return;
         }
-        else if (state.getClass().equals(Accepted.class))
-        {
 
-        }
-        else if (state.getClass().equals(Rejected.class))
-        {
-            Rejected rejectedState = (Rejected)state;
+        if (state.getClass().equals(Rejected.class)) {
```

Comment:
Is this else if --> if change for some warning here?

Reply:
the warning was again that all these else if blocks contained empty body. 

The reason I chose to modify the if-else checks is to separate 2 cases
 -> (state == null) => assign an exception and return
-> statue is not null => perform subsequent condition checks

previously we have all the checks being performed in a single linear flow.